text_mining.py

```diff
@@ -0,0 +1,357 @@
+    """
+    Returns a list of previous pickled user instances of EPL players
+    """
+    f = open(filename, 'rb')
+    players = []
+    num_players = load(f)  # loads the number of players
+    try:
+        for i in range(num_players):
+            players.append(load(f))
+    finally:
+        f.close()
+        return players
+
+
+def save_tweets(player_id, player_name, api):
+    """
+    Pickles file "players_status"/player_name.pickle" of
+    200 latest Status Instances of player with player_id.
+    200 is the maximum number of tweets limited by Twitter Api.
+    """
+    #  pickles the latest 200(max limit) statuses for each player
+    statuses = api.GetUserTimeline(user_id=player_id, count=200)
+    if not exists("players_status"):
+        mkdir("players_status")
+    f = open("players_status/" + player_name, 'wb')
+    dump(len(statuses), f)  # saves the number of total Tweets
+    try:
+        for status in statuses:
+            dump(status, f)
+    finally:
+        f.close()
+
+
+def load_tweets(player_name):
+    """
+    Returns a list of all pickled Status Instances of player with player_name
+    """
+    f = open("players_status/" + player_name, 'rb')
+    statuses = []
+    num_status = load(f)  # loads the number of players
+    try:
+        for i in range(num_status):
+            statuses.append(load(f))
+    finally:
+        f.close()
+        return statuses
+
+
+def get_texts_from_player(player_name, start_date):
+    """
+    Returns only texts that were created after time object "start_date"
+    from a player(identified with player_name)'s statuses
+    from the pickled files.
+    Eliminates Tags('@'), Hashtags(#)
+    and URLs that are irrelevant to text sentiment.
+    """
+    statuses = load_tweets(player_name)
+    player_texts = []
+    for status in statuses:
+        status_time = datetime.datetime.strptime(status.created_at,
+                                                 '%a %b %d %X %z %Y')
+        if status_time >= start_date:
+            #  regular expression that eliminates URL and @,
+            #  source :
+            #  http://stackoverflow.com/questions/8376691/
+            #  how-to-remove-hashtag-user-link-of-a-tweet-using-regular-expression
+            text = status.text
+            text = re.sub("(@[A-Za-z0-9]+)|([^0-9A-Za-z \t])|(\w+:\/\/\S+)",
+                          "", text)
+            player_texts.append(text)
+    return player_texts
+
+
+def save_tweets_texts(filename, players):
+    """
+    Pickles the collection of refined texts for players as a dictionary.
+    in file (filename)
+    """
+    tweet_texts = dict()  # dictionary of all refined texts.
+    # initializing the dictionary
+    for player in players:
+        if not player.protected:
+            try:
+                tweet_texts[player.name] = get_texts_from_player(player.name,
+                                                                 START_TIME)
+            except Exception:
```

Comment:
This is very dangerous to do. `except Exception` is usually never good idea because it ignores all errors that you might needed to fix!